hedera-node/src/main/java/com/hedera/services/contracts/operation/HederaSLoadOperation.java

```diff
@@ -96,9 +96,10 @@ public OperationResult execute(final MessageFrame frame, final EVM evm) {
                         optionalCost, Optional.of(ExceptionalHaltReason.INSUFFICIENT_GAS));
             } else {
                 UInt256 storageValue = account.getStorageValue(UInt256.fromBytes(key));
-                if (dynamicProperties
-                        .enabledSidecars()
-                        .contains(SidecarType.CONTRACT_STATE_CHANGE)) {
+                if (!frame.isStatic()
```

Comment:
Why do we want to turn off sload for static calls?  If it's a local call the changes to `EvmTxProcessor` should keep the tracer from being turned on.  If it's not a local call we will still need to track state reads for future re-execution validation.

Reply:
Isn't it guarding only the call to [HederaOperationUtil.cacheExistingValue](https://github.com/hashgraph/hedera-services/blob/388f56548b0bda5a03cf4301f3e6f4dfe2ca1d62/hedera-node/src/main/java/com/hedera/services/contracts/operation/HederaOperationUtil.java#L172-L193), which only populates the slot as read in the[ `stateChanges` in HederaWorldState.Updater](https://github.com/hashgraph/hedera-services/blob/master/hedera-node/src/main/java/com/hedera/services/store/contracts/HederaWorldState.java#L182), which are only used for sidecars? And since sidecars are not externalized with queries, these calls and the tracking of state changes are kind of unnecessary. 

The actions and state changes are added to the `TransactionProcessingResult`, but they do not get externalized in the [CallLocal scenario](https://github.com/hashgraph/hedera-services/blob/388f56548b0bda5a03cf4301f3e6f4dfe2ca1d62/hedera-node/src/main/java/com/hedera/services/contracts/execution/CallLocalExecutor.java#L101), since the response protobuf does not contain fields for them.

> Rather than turning tracing off we would want to change the output to be part of the return protobuf for the request.

I don't think there will be any substantial performance gains from the changes in this PR, so if we are planning to do the above anyway, we can probably skip this, since we will be re-enabling and reverting the changes from this PR when the time comes to implement it? Not sure what the timeline looks like here or what your thoughts are on this?

Reply:
@dimitar-dinev base on the above you should be able to remove this and the PR will be a lot simpler

Reply:
> There will be static call frames in non-local calls. It's a standard part of how solidity compiles view and pure functions. It will show up in `ContractCall` and `ContractCallLocal` calls and needs to be tracked in the side car.
🤦🏽  Of course, will remove

Reply:
only the `!isStatic` was supposed to be removed.  We still need the enabled sidecars check.

```
            if (dynamicProperties.enabledSidecars().contains(CONTRACT_STATE_CHANGE)) {
                cacheExistingValue(frame, address, key, account.getStorageValue(key));
            }
```